sources/web/datalab/auth.ts

```diff
@@ -145,24 +169,22 @@ export function handleAuthFlow(request: http.ServerRequest, response: http.Serve
         }
       });
     }
-  } else if (!oauth2Client && process.env.DATALAB_ENV == 'local' && !fs.existsSync(appCredFile)) {
+  } else if (!oauth2Client && !isSignedIn()) {
```

Comment:
Glad you suggested that as I found an issue as a result. We need to handle the race condition of sign in being clicked twice, so if we are already signed in need to just redirect back to referer.